# Search Trends Data Analysis

The dataset contains historical search trends from different locations between 2001 and 2020, showing the top-ranked search queries across various categories.

In [35]:
import pandas as pd
import numpy as np

In [36]:
df = pd.read_csv('trends.csv')
df

,location,year,category,rank,query
0,Global,2001,Consumer Brands,1,Nokia
1,Global,2001,Consumer Brands,2,Sony
2,Global,2001,Consumer Brands,3,BMW
3,Global,2001,Consumer Brands,4,Palm
4,Global,2001,Consumer Brands,5,Adobe
...,...,...,...,...,...
26950,Vietnam,2020,Là Gì?,1,Virus Corona là gì
26951,Vietnam,2020,Là Gì?,2,Miễn thị thực là gì
26952,Vietnam,2020,Là Gì?,3,Đầu cắt moi là gì
26953,Vietnam,2020,Là Gì?,4,Bệnh bạch hầu là gì


## Initial Analysis

In [37]:
df.shape

(26955, 5)

In [38]:
df.head()

,location,year,category,rank,query
0,Global,2001,Consumer Brands,1,Nokia
1,Global,2001,Consumer Brands,2,Sony
2,Global,2001,Consumer Brands,3,BMW
3,Global,2001,Consumer Brands,4,Palm
4,Global,2001,Consumer Brands,5,Adobe


In [39]:
df.tail()

,location,year,category,rank,query
26950,Vietnam,2020,Là Gì?,1,Virus Corona là gì
26951,Vietnam,2020,Là Gì?,2,Miễn thị thực là gì
26952,Vietnam,2020,Là Gì?,3,Đầu cắt moi là gì
26953,Vietnam,2020,Là Gì?,4,Bệnh bạch hầu là gì
26954,Vietnam,2020,Là Gì?,5,Đông Lào là gì


In [40]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 26955 entries, 0 to 26954
Data columns (total 5 columns):
 #   Column    Non-Null Count  Dtype 
---  ------    --------------  ----- 
 0   location  26955 non-null  object
 1   year      26955 non-null  int64 
 2   category  26955 non-null  object
 3   rank      26955 non-null  int64 
 4   query     26955 non-null  object
dtypes: int64(2), object(3)
memory usage: 1.0+ MB


In [41]:
df.describe()

,year,rank
count,26955.000000,26955.00000
mean,2015.243369,3.00000
std,3.564683,1.41424
min,2001.000000,1.00000
25%,2013.000000,2.00000
50%,2016.000000,3.00000
75%,2018.000000,4.00000
max,2020.000000,5.00000


## Data Cleaning

In [42]:
df.columns

Index(['location', 'year', 'category', 'rank', 'query'], dtype='object')

In [43]:
df.dtypes

location    object
year         int64
category    object
rank         int64
query       object
dtype: object

### Checking for Missing Values

In [44]:
df.isnull().sum()

location    0
year        0
category    0
rank        0
query       0
dtype: int64

### Removing Duplicate Values

In [45]:
df.duplicated().sum()

np.int64(10)

In [46]:
# Removing the 10 duplicate rows
df = df.drop_duplicates()

In [47]:
# Verifying removal of duplicates
df.duplicated().sum()

np.int64(0)

In [48]:
df.nunique()

location       83
year           20
category     2450
rank            5
query       18431
dtype: int64

## Statistical Analysis

In [49]:
df.mean(numeric_only=True)

year    2015.241974
rank       3.000000
dtype: float64

In [50]:
df.median(numeric_only=True)

year    2016.0
rank       3.0
dtype: float64

In [51]:
df.min(numeric_only=True)

year    2001
rank       1
dtype: int64

In [52]:
df.max(numeric_only=True)

year    2020
rank       5
dtype: int64

In [53]:
df.std(numeric_only=True)

year    3.564557
rank    1.414240
dtype: float64

## Overall Interpretation

- The dataset contains 26,955 records and 5 columns representing location, year, category, rank, and query.
- The dataset contains no missing values, while 10 duplicate records were identified.
- The year values range from 2001 to 2020 and the rank values range from 1 to 5, with no obvious invalid values observed.
- The dataset contains 83 unique locations, 2,450 unique categories, and 18,431 unique queries.
- Overall, the dataset is relatively complete and requires minimal cleaning, with duplicate records being the main issue identified.